src/Mailgun/Api/Domain.php

```diff
@@ -102,15 +106,15 @@ public function create($domain, $smtpPass, $spamAction, $wildcard)
      *
      * @param string $domain Name of the domain.
      *
-     * @return SimpleResponse|array|ResponseInterface
+     * @return DeleteResponse|array|ResponseInterface
      */
-    public function remove($domain)
+    public function delete($domain)
     {
         Assert::stringNotEmpty($domain);
 
-        $response = $this->delete(sprintf('/v3/domains/%s', $domain));
+        $response = $this->httpDelete(sprintf('/v3/domains/%s', $domain));
```

Comment:
I had to rename this function to avoid recursion... Maybe we should rename all protected HTTP method to httpGet, httpPost etc?